sources/packages/backend/libs/services/src/health-check/health.service.ts

```diff
@@ -0,0 +1,74 @@
+import { Injectable } from "@nestjs/common";
+import { RedisOptions, Transport } from "@nestjs/microservices";
+import {
+  HealthIndicator,
+  HealthIndicatorResult,
+  HealthCheckError,
+  MicroserviceHealthIndicator,
+} from "@nestjs/terminus";
+import { ConfigService } from "@sims/utilities/config";
+import { ZeebeHealthIndicator } from "./zeebe-health-indicator";
+
+@Injectable()
+export class HealthService extends HealthIndicator {
+  constructor(
+    private microservice: MicroserviceHealthIndicator,
+    private readonly configService: ConfigService,
+    private readonly zeebeHealthIndicator: ZeebeHealthIndicator,
+  ) {
+    super();
+  }
+
+  /**
+   * Check the health of the application.
+   * @param key specifies if the health check is done for workers/queues.
+   * @returns status of the health check.
+   * In case of queues, we check if the redis is up and running.
+   * In case of workers, we check if the zeebe is ready to accept the connections.
+   */
+  async isHealthy(key: string): Promise<HealthIndicatorResult> {
+    let isHealthy;
+    let healthCheckResult;
```

Comment:
I see `healthCheckResult` in lines 34 and 37 are `const`, then what is the use of this `let` variable, and won't the `healthCheckResult` in line 42 will be always `undefined`?